wordpress-mobile/WordPress-iOS
# Share Extension: Fix for posts published with local paths

Fixes #9118

Sometimes the images in a post created with the share extension would keep their local paths once published. This PR should fix those problems.

This was a fun one to track down: It seems that when a media's upload is interupted the server may alter the file's name, such as turning "localfile.jpg" into "localfile-1.jpg". I assume this is related to having to reattempt the upload but I wasn't able to be sure of that. Regardless I was able to fix the issue with that assumption by making the filename comparison a bit more forgiving.

### To test:
Perform the text before and after this fix to experience the issue, and then ensure it has been addressed:
1. Build to device
2. Share two images with the WordPress share extension.
3. Immediately after tapping "Publish" turn off wifi (and LTE if you have it) on the device to interupt media upload
4. Wait a bit, then turn wifi back off.
5. Compare results:
   - Before the fix, the post should have broken media [like this](https://natesnotesandnovels.wordpress.com/2019/04/26/169/)
   - After the fix, embedded images should [appear correctly](https://natesnotesandnovels.wordpress.com/2019/04/26/201/)

Update release notes:
